citesphere/src/main/java/edu/asu/diging/citesphere/core/authority/impl/ViafAuthorityImporter.java

```diff
@@ -93,13 +122,98 @@ public String getId() {
+    public AuthoritySearchResult searchAuthorities(String firstName, String lastName, int page, int pageSize)
+            throws AuthorityServiceConnectionException {
+
+        // VIAF search API response contains maximum of 10 records
+        pageSize = Math.min(10, pageSize);
+        String url = viafURL + viafSearchKeyword + this.getViafSearchString(firstName, lastName, page, pageSize);
+
+        ResponseEntity<ViafSearchResponseWrapper> response = null;
+        try {
+            response = restTemplate.exchange(url, HttpMethod.GET, null, ViafSearchResponseWrapper.class);
+        } catch (RestClientException ex) {
+            throw new AuthorityServiceConnectionException(ex);
+        }
+
+        List<IAuthorityEntry> authorityEntries = new ArrayList<IAuthorityEntry>();
+        AuthoritySearchResult searchResult = new AuthoritySearchResult();
+        if (response.getStatusCode() == HttpStatus.OK) {
+            ViafSearchResponse viafSearchResponse = response.getBody().getSearchRetrieveResponse();
+            if (viafSearchResponse.getRecords() != null) {
+                for (ViafSearchItem viafEntry : viafSearchResponse.getRecords()) {
+                    ViafRecordData record = viafEntry.getRecord().getRecordData();
+                    IAuthorityEntry authority = new AuthorityEntry();
+                    Iterator<Data> iterator = record.getMainHeadings().getData().iterator();
+                    /* MainHeading data may contain multiple items in some cases. Using the first item for fetching the display name.
+                     * For e.g.,
+                     * "data": [
+                                {
+                                    "text": "Albert, August 1881-",
+                                    "sources": {
+                                        "s": "DNB",
+                                        "sid": "DNB|1204691185"
+                                    }
+                                },
+                                {
+                                    "text": "August Albert German philologist",
+                                    "sources": {
+                                        "s": "WKP",
+                                        "sid": "WKP|Q83561432"
+                                    }
+                                }
+                            ],
+                     */
+                    if (iterator.hasNext()) {
+                        String name = iterator.next().getText();
+                        authority.setName(name);
+                        authority.setUri(record.getDocument().get("@about") + "");
+                        authorityEntries.add(authority);
+                    }
+                }
+                searchResult.setFoundAuthorities(authorityEntries);
+                searchResult
+                        .setTotalPages((int) Math.ceil(viafSearchResponse.getNumberOfRecords() / new Float(pageSize)));
+            }
+        } else {
```

Comment:
I believe by default the resttemplate throws an exception when there is no success return code. I don't think the default behavior has changed here?